languages/python/concepts/list-methods/about.md

```diff
@@ -0,0 +1,286 @@
+
+```python
+>>> numbers = [1, 2, 3]
+>>> numbers.clear()
+>>> numbers
+[]
+```
+
+## Reverse the list
+
+The items in the list can be reordered in the reverse order with the `reverse` function.
+
+```python
+>>> numbers = [1, 2, 3]
+>>> numbers.reverse()
+>>> numbers
+[3, 2, 1]
+```
+
+## Sort the list
+
+When you have an unordered list of items, you can sort them with the help of `sort` function. If you have items that are alphanumerical, you can use `sort` without any arguments.
+
+If you have a list of names and you want to sort them:
+
+```python
+>>> names = ["Tony", "Natasha", "Thor", "Bruce"]
+>>> names.sort()
+>>> names
+["Bruce", "Natasha", "Thor", "Tony"]
+```
+
+If you want the sort to be in descending order, you can use the reverse parameter:
+
+```python
+>>> names = ["Tony", "Natasha", "Thor", "Bruce"]
+>>> names.sort(reverse=True)
+>>> names
+["Tony", "Thor", "Natasha", "Bruce"]
+```
+
+If you have a list of items that are complex, you can use the key parameter in the sort. Lets see more about this later.
+
+Internally, python uses [`Timsort`][timsort] to sort the list.
+
+## Occurrences of an item in the list
+
+You can find the occurrences of an element in the list with the help of the `count` function. It takes one parameter which will be the item, for which you need to find the number of occurrences.
+
+```python
+>>> items = [1, 4, 7, 8, 2, 9, 2, 1, 1, 0, 4, 3]
+>>> items.count(1)
+3
+```
+
+## Find the index of the item
+
+If you want to find the index of an item you want in the list, you can use the `index` function. If you have multiple occurrences of the same item, it will provide you the index of the first occurrence. If you do not have any occurrences of the item, then you will have a `ValueError` raised. The index is zero-based, meaning the first item is at the position 0.
+
+```python
+>>> items = [7, 4, 1, 0, 2, 5]
+>>> items.index(4)
+1
+>>> items.index(10)
+ValueError: 10 is not in list
+```
+
+You can also provide a start (and an end) to the index from where you need the search has to happen for the item.
+
+```python
+>>> names = ["Tina", "Leo", "Thomas", "Tina", "Emily", "Justin"]
+>>> names.index("Tina")
+0
+>>> names.index("Tina", 2, 5)
+3
+```
+
+## Copy a List
+
+Lists are collections of items which are referenced by an index. Therefor, if you do re-assign a list object to another variable, any change you do to the new variable will also impact the original variable.
+
+```python
+>>> actual_names = ["Tony", "Natasha", "Thor", "Bruce"]
+>>> same_list = actual_names
+>>> same_list.append("Clarke")
+>>> same_list
+["Tony", "Natasha", "Thor", "Bruce", "Clarke"]
+>>> actual_names
+["Tony", "Natasha", "Thor", "Bruce", "Clarke"]
+```
+
+You need to use the `copy` function in order to do a `shallow_copy` of the list. (More about `shallow_copy` and `deep_copy` later.)
```

Comment:
```suggestion
To avoid this issue,  use the `list.copy()` method to make a `shallow_copy` of the list. `copy()` will create a new list object, but will not create new objects for the list items. (More about the differences between a  `shallow_copy` and  a `deep_copy` later.)
```